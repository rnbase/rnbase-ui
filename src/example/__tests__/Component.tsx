import React from 'react';

// Note: test renderer must be required after react-native.
import TestRenderer from 'react-test-renderer';

/**
 * Under test
 */
import Component, { Props } from '../Component';

const createElement = (props: Props) => (
  <Component {...props}>TEXT</Component>
);

const createRenderer = (props: Props) => TestRenderer.create(createElement(props));

it('should render normally', () => {
  expect(createRenderer({})).toMatchSnapshot();
});

it('should not render', () => {
  expect(createRenderer({ visible: false })).toMatchSnapshot();
});

it('should render with custom styles', () => {
  expect(
    createRenderer({
      style: { backgroundColor: '#FFFFFF' },
      textStyle: { color: '#000000' },
    }),
  ).toMatchSnapshot();
});
