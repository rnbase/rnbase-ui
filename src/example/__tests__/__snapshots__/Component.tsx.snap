// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render 1`] = `null`;

exports[`should render normally 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#AEAEB2",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "paddingHorizontal": 15,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": "bold",
        "textTransform": "uppercase",
      }
    }
  >
    TEXT
  </Text>
</View>
`;

exports[`should render with custom styles 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "paddingHorizontal": 15,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": "bold",
        "textTransform": "uppercase",
      }
    }
  >
    TEXT
  </Text>
</View>
`;
